civis/tests/test_io.py

```diff
@@ -129,11 +132,13 @@ def test_bytes_file_to_civis(self, *mocks):
     def test_large_file_to_civis(self, *mocks):
         curr_size = civis.io._files.MIN_MULTIPART_SIZE
         civis.io._files.MIN_MULTIPART_SIZE = 1
-        with tempfile.NamedTemporaryFile() as tmp:
-            tmp.write(b'a,b,c\n1,2,3')
-            tmp.flush()
-            tmp.seek(0)
-            result = civis.io.file_to_civis(tmp, tmp.name)
+        with TemporaryDirectory() as temp_dir:
+            fname = os.path.join(temp_dir, str(uuid.uuid4()))
+            with open(fname, 'w+b') as tmp:
+                tmp.write(b'a,b,c\n1,2,3')
+                tmp.flush()
+                tmp.seek(0)
```

Comment:
I would exit the write context and re-open the file in a new `with open(...)` context here. The `seek(0)` was so that we could use a temporary file which disappeared when closed.